software/sw_build.mk

```diff
@@ -63,14 +72,22 @@ TEMPLATE_LDS=src/$@.lds
 
 IOB_SOC_OPENCRYPTOLINUX_CFLAGS ?=-Os -nostdlib -march=rv32imac -mabi=ilp32 --specs=nano.specs -Wcast-align=strict
 
-IOB_SOC_OPENCRYPTOLINUX_INCLUDES=-I. -Isrc 
+IOB_SOC_OPENCRYPTOLINUX_INCLUDES=-I. -Isrc -Isrc/crypto/McEliece -Isrc/crypto/McEliece/common
 
 IOB_SOC_OPENCRYPTOLINUX_LFLAGS=-Wl,-Bstatic,-T,$(TEMPLATE_LDS),--strip-debug
 
 # FIRMWARE SOURCES
 IOB_SOC_OPENCRYPTOLINUX_FW_SRC=src/iob_soc_opencryptolinux_firmware.S
 IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=src/iob_soc_opencryptolinux_firmware.c
 IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=src/printf.c
+
+# NOTE(Ruben): To speed up simulation, the following files can be commented out to greatly reduce filesize from around ~120Kb to ~20Kb.
+IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=src/versat_crypto.c
+IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=src/versat_crypto_tests.c
+IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=src/crypto/aes.c
+IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=$(wildcard src/crypto/McEliece/*.c)
+IOB_SOC_OPENCRYPTOLINUX_FW_SRC+=$(wildcard src/crypto/McEliece/common/*.c)
```

Comment:
sugestion: add only to `sw_build.mk` if we are using versat? Use a setup like `USE_ETHERNET`?

Reply:
Did everything except this part. How do I approach this in the correct python setup approach? I do not see any variable inside the Makefile that indicates we are simulating vs running fpga. Also do not see any direct way of telling on the python side, so cannot conditionally insert the lines in sw_build because cannot know if doing setup to sim or doing setup to run fpga.

Reply:
An ideal solution would be to remove unused code from compilation.
After [some search](https://stackoverflow.com/a/6770305), it seems that adding the compilation and linking flags: `-fdata-sections -ffunction-sections -Wl,--gc-sections` would remove unused functions.
But it yields an elf without sections:
```bash
riscv64-unknown-elf-objcopy: error: the input file 'iob_soc_opencryptolinux_boot.elf' has no sections
```

I think we should just keep this as is.

Reply:
> sugestion: add only to `sw_build.mk` if we are using versat? Use a setup like `USE_ETHERNET`? or skip sources for simulation flows

I've added a way to skip sources for simulation in [a suggestion below](https://github.com/IObundle/iob-soc-opencryptolinux/pull/82#discussion_r1506863775), but I have not tried to run it, so it may break compilation if the firmware requires those functions to be present.